app/src/main/java/at/roteskreuz/stopcorona/model/repositories/QuarantineRepository.kt

```diff
@@ -380,8 +381,22 @@ class QuarantineRepositoryImpl(
     override fun receivedWarning(warningType: WarningType, timeOfContact: ZonedDateTime) {
         @Suppress("NON_EXHAUSTIVE_WHEN")
         when (warningType) {
-            WarningType.YELLOW -> dateOfLastYellowContact = timeOfContact
-            WarningType.RED -> dateOfLastRedContact = timeOfContact
+            WarningType.YELLOW -> {
+                if (dateOfLastYellowContact == null || dateOfLastYellowContact!!.areOnTheSameDay(timeOfContact).not()) {
+                    dateOfLastYellowContact = timeOfContact
+                } else {
+                    Timber.d("no update of the yellow quarantine neccesary, we stay at " +
```

Comment:
`else`-logging could be removed in the next task

Reply:
I will leave it in. This was very concious as the whole process is entirely in the background and Exceptions are cought silently. In case we have some problems, we can process the debug log.